src/lib/genericEmbedPagination.ts

```diff
@@ -50,7 +50,8 @@ export async function createPaginatedEmbedHandler(initialInteraction: CommandInt
     }
 
     function handleEmbedExpire() {
-        initialInteraction.editReply({ embeds: [embeds[currentPage].setFooter({ text: 'This embed has expired.' })], components: [] });
+        const embed = embeds[currentPage];
+        initialInteraction.editReply({ embeds: [embed.setFooter({ text: `${embed.data.footer ? `${embed.data.footer.text} - ` : ''} This embed has expired.` })], components: [] });
```

Comment:
```suggestion
        initialInteraction.editReply({ embeds: [embed.setFooter({ text: `${embed.data.footer ? `${embed.data.footer.text} - ` : ''}This embed has expired.` })], components: [] });
```

My bad, I left a space here resulting in a double space when coupled with the `-`